Ollivanders/src/net/pottercraft/Ollivanders2/O2Player.java

```diff
@@ -307,6 +336,39 @@ public void setSpellCount (Spells spell, int count)
       }
    }
 
+   /**
+    * Set the most recent cast time for a spell. This will override the existing values for this spell.
+    *
+    * @param spell the spell to set the time for
+    */
+   public void setSpellRecentCastTime (Spells spell)
+   {
+      String spellClass = "net.pottercraft.Ollivanders2.Spell." + spell.toString();
```

Comment:
In general I would like to see a better way of getting the spell than doing string concatenation and reflection.  Problems with the spellClass string not matching the classname won't show up until runtime when the specific spell is attempted.